Advanced-Recruit-Training/src/BatteryStateOfChargeService.cpp

```diff
@@ -1,32 +1,50 @@
 #include "BatteryStateOfChargeService.h"
+#include "BatteryData.h"
 
 namespace
 {
     const double BATTERY_AMP_HOUR_CAPACITY = 123.0;
+    const double MS_IN_HOUR = 3600000;
 }
 
 BatteryStateOfChargeService::BatteryStateOfChargeService(double initialStateOfChargePercent)
 : initialStateOfChargePercent_(initialStateOfChargePercent)
 {
+    ampHoursUsed_ = BATTERY_AMP_HOUR_CAPACITY * initialStateOfChargePercent_ / 100;
 }
 
+// Empty since children will be destroyed
 BatteryStateOfChargeService::~BatteryStateOfChargeService()
 {
 }
 
 double BatteryStateOfChargeService::totalAmpHoursUsed() const
 {
-    return 0.0;
+    return ampHoursUsed_;
 }
 
 bool BatteryStateOfChargeService::isCharging() const
 {
+    if (dataPoint_.current < 0) {
+       return true;
+    }
     return false;
 }
 
+// Amphours used divided by current or remaining amphours divided by current
 QTime BatteryStateOfChargeService::timeWhenChargedOrDepleted() const
 {
-    return QTime::currentTime();
+    double hoursTillChargedOrDepleted;
+    QTime msDuration(0,0,0,0);
```

Comment:
I feel like these two names should be switched, since `msDuration` isn't actually in milliseconds but is a QTime that says the time until charged or depleted, and `hoursTillChargedOrDepleted` gets turned into the duration in ms later!